python/ray/util/scheduling_strategies.py

```diff
@@ -72,9 +72,95 @@ def __init__(
         self._fail_on_unavailable = _fail_on_unavailable
 
 
+def _validate_values(values):
+    if values is None or len(values) == 0:
+        raise ValueError("The value of label cannot be empty.")
+
+    index = 0
+    for value in values:
+        if not isinstance(value, str):
+            raise ValueError(
+                f"Expected value {str(value)} in position {index} to be of type str"
+            )
+        index = index + 1
+
+
+class In:
+    def __init__(self, *values):
+        _validate_values(values)
+        self.values = list(values)
+
+
+class NotIn:
+    def __init__(self, *values):
+        _validate_values(values)
+        self.values = list(values)
+
+
+class Exists:
+    def __init__(self):
+        pass
+
+
+class DoesNotExist:
+    def __init__(self):
+        pass
+
+
+class _LabelMatchExpression:
+    """An expression used to select node by node's labels
+    Attributes:
+        key: the key of label
+        operator: In、NotIn、Exists、DoesNotExist
+    """
+
+    def __init__(self, key: str, operator: Union[In, NotIn, Exists, DoesNotExist]):
+        self.key = key
+        self.operator = operator
+
+
+LabelMatchExpressionsT = Dict[str, Union[In, NotIn, Exists, DoesNotExist]]
+
+
+@PublicAPI(stability="alpha")
+class NodeLabelSchedulingStrategy:
+    """Label based node affinity scheduling strategy
+
+    scheduling_strategy=NodeLabelSchedulingStrategy({
+          "region": In("us"),
+          "gpu_type": Exists()
+    })
+    """
+
+    def __init__(
+        self, hard: LabelMatchExpressionsT, *, soft: LabelMatchExpressionsT = None
+    ):
+        self.hard = _convert_map_to_expressions(hard, "hard")
+        self.soft = _convert_map_to_expressions(soft, "soft")
+
+
+def _convert_map_to_expressions(map_expressions: LabelMatchExpressionsT, param: str):
+    expressions = []
+    if map_expressions is None:
+        return expressions
+
+    for key, value in map_expressions.items():
+        if isinstance(value, (In, NotIn, Exists, DoesNotExist)):
+            expressions.append(_LabelMatchExpression(key, value))
+        else:
+            raise ValueError(
+                f"Expected the value of key `{key}` in "
+                f"`{param}` label match expression of "
+                "node label scheduling strategy to be of type "
+                "`In` or `NotIn` or`Exists` or `DoesNotExist`."
```

Comment:
The map value of the {param} parameter must be one of the `In`, `NotIn`, `Exists` or `DoesNotExist` operator (e.g. {"key1": In("value1")}) but got {value} of type {type(value)}